tests/blackbox_test.go

```diff
@@ -259,13 +261,22 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	// If we're not expecting the config to be bad, make sure it passes
 	// validation.
 	if !test.ConfigShouldBeBad {
-		_, rpt, err := config.Parse([]byte(test.Config))
+		renderedConfig, rpt, err := config.Parse([]byte(test.Config))
 		if rpt.IsFatal() {
 			return fmt.Errorf("test has bad config: %s", rpt.String())
 		}
 		if err != nil {
 			return fmt.Errorf("error parsing config: %v", err)
 		}
+		defer func() {
+			if test.Config != "" {
```

Comment:
Do we even need this gate?

Reply:
I guess not, I know at one point I was running into an empty config but it was probably circumstantial due to something else that was wrong.. I have removed this.